File: test-ghost-text.js
// xCopilot Ghost Text Test File
// Test various scenarios for ghost text suggestions

// Test 1: Simple function declaration
function calculateSum(a, b) {
    // Cursor here should suggest: return a + b;
}

// Test 2: Conditional statements  
function checkUser(user) {
    if (user && user.isActive) {
        // Cursor here should suggest appropriate action
    }
    
    // Cursor here should suggest else block or next statement
}

// Test 3: Loop structures
function processItems(items) {
    for (let i = 0; i < items.length; i++) {
        // Cursor here should suggest processing logic
    }
    
    // Cursor here should suggest return or next logic
}

// Test 4: Object/Class structures
class UserService {
    constructor() {
        // Cursor here should suggest initialization
    }
    
    async fetchUser(id) {
        // Cursor here should suggest async logic
    }
    
    // Cursor here should suggest new methods
}

// Test 5: Variable declarations
const users = data.filter(
    // Cursor here should suggest filter predicate
);

const result = fetch('/api/users').then(
    // Cursor here should suggest promise handling
);

// Test 6: Console/logging
console.log(
    // Cursor here should suggest log content
);

// Test 7: Error handling
try {
    // Cursor here should suggest try block content
} catch (error) {
    // Cursor here should suggest error handling
}

// Test 8: Arrow functions
const processData = (data) => {
    // Cursor here should suggest function body
};

// Test 9: Async/await patterns
async function loadData() {
    // Cursor here should suggest await patterns
}

// Test 10: Import/Export (if detected)
// Cursor here should suggest imports or exports

/*
Instructions for testing:
1. Open this file in VS Code with xCopilot extension enabled
2. Place cursor at the end of comment lines mentioning "Cursor here"
3. Wait 500ms for ghost text to appear
4. Press Tab to accept or Esc to reject
5. Verify suggestions are contextually appropriate
6. Test multi-line suggestions for functions and classes
7. Test hover tooltips if preview is enabled
8. Test Ctrl+Alt+G to toggle ghost text on/off
*/

File: test-ghost-text.py
# xCopilot Ghost Text Test File - Python
# Test various Python scenarios for ghost text suggestions

# Test 1: Function definitions
def calculate_sum(a, b):
    # Cursor here should suggest: return a + b

def process_data(df):
    # Cursor here should suggest pandas operations or data processing

# Test 2: Class definitions
class UserService:
    def __init__(self):
        # Cursor here should suggest initialization
    
    def get_user(self, user_id):
        # Cursor here should suggest database query or API call
    
    # Cursor here should suggest new methods

# Test 3: Conditional statements
def check_user(user):
    if user and user.get('active'):
        # Cursor here should suggest appropriate action
    
    # Cursor here should suggest else clause

# Test 4: Loop structures  
def process_items(items):
    for item in items:
        # Cursor here should suggest item processing
    
    # Cursor here should suggest return or next logic

# Test 5: List comprehensions
users = [user for user in data if 
    # Cursor here should suggest filter condition
]

# Test 6: Dictionary operations
user_data = {
    'id': user_id,
    'name': user_name,
    # Cursor here should suggest more fields
}

# Test 7: Exception handling
try:
    # Cursor here should suggest try block content
except Exception as e:
    # Cursor here should suggest error handling

# Test 8: Async functions (Python 3.7+)
async def fetch_data():
    # Cursor here should suggest await patterns

# Test 9: Lambda functions
process_func = lambda x: 
    # Cursor here should suggest lambda body

# Test 10: Import statements
from datetime import 
    # Cursor here should suggest datetime imports

# Test 11: With statements
with open('file.txt', 'r') as file:
    # Cursor here should suggest file operations

"""
Instructions for testing Python ghost text:
1. Open this file in VS Code with xCopilot extension enabled
2. Place cursor at the end of comment lines mentioning "Cursor here"
3. Wait 500ms for ghost text to appear
4. Verify Python-specific suggestions appear
5. Test multi-line function and class suggestions
6. Test context-aware suggestions based on function names
7. Verify language-specific patterns are detected
"""